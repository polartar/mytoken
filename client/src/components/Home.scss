.home {
  padding-bottom: 40px;
  button.btn.tokenize-now {
    font-size: 1.5em;
    padding: 10px 20px;
  }
}