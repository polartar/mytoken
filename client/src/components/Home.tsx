import React from 'react';
import { connect } from 'react-redux';
import {
  useHistory,
} from 'react-router-dom';

import {
  Card,
  Button,
} from 'react-bootstrap';

import Content from '../shared/components/Content';
import { TokenizeActions, tokenizeActions } from '../redux/tokenize';

import './Home.scss';

type Props = TokenizeActions;

const Home = (props: Props) => {
  const { resetTokenDetails } = props;
  const history = useHistory();

  const tokenizeNow = () => {
    resetTokenDetails();
    history.push( '/tokenize' );
  };

  return (
    <div className='home'>
      <Content />
      <Card style={ { border: 'none', background: '#f8faff', margin: '20px 10px' } }>
        <h3 className='mt-5'>Start Tokenizing Your Assets Today</h3>
      </Card>
      <Button className='tokenize-now' variant='primary' onClick={ tokenizeNow }>Tokenize Now!</Button>
    </div>
  );
};

export { Home as PureHome }; // Pure component, for testing
export default connect( null, { ...tokenizeActions } )( Home );
