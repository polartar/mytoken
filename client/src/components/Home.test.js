import React from 'react';
import renderer from 'react-test-renderer';

import { PureHome as Home } from './Home';

describe('Home', () => {
  let props;

  beforeEach(() => {
    props = {
      resetTokenDetails: jest.fn(),
    };
  });
  
  it('renders without crashing', () => {
    const component = renderer.create(<Home {...props}/>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
