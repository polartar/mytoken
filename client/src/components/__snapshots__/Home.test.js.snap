// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
<div
  className="home"
>
  <div
    className="carousel"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="active carousel-item"
      >
        <div
          className="carousel-caption"
        >
          <div
            className="title"
          >
            <h1>
              Welcome to the Self Tokeninzation Platform
            </h1>
            <h3
              className="mt-4"
            >
              Securely Tokenize your assets on the blockchain
            </h3>
          </div>
          <h2
            className="mt-5"
          >
            Tokenize real world assets
          </h2>
          <h4
            className="mb-5"
          >
            Securely and transparently
          </h4>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <div
          className="carousel-caption"
        >
          <div
            className="title"
          >
            <h1>
              Welcome to the Self Tokeninzation Platform
            </h1>
            <h3
              className="mt-4"
            >
              Securely Tokenize your assets on the blockchain
            </h3>
          </div>
          <h2
            className="mt-5"
          >
            From real estate to art and more
          </h2>
          <h4
            className="mb-5"
          >
            Safely and easily harness the benefits of blockchain
          </h4>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <div
          className="carousel-caption"
        >
          <div
            className="title"
          >
            <h1>
              Welcome to the Self Tokeninzation Platform
            </h1>
            <h3
              className="mt-4"
            >
              Securely Tokenize your assets on the blockchain
            </h3>
          </div>
          <h2
            className="mt-5"
          >
            Broaden the investment base for your assets
          </h2>
          <h4
            className="mb-5"
          >
            Improve liquidity through a frictionless environment
          </h4>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <div
    className="card"
    style={
      Object {
        "background": "#f8faff",
        "border": "none",
        "margin": "20px 10px",
      }
    }
  >
    <h3
      className="mt-5"
    >
      Start Tokenizing Your Assets Today
    </h3>
  </div>
  <button
    className="tokenize-now btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Tokenize Now!
  </button>
</div>
`;
